server/src/main/java/org/apache/cloudstack/user/PasswordResetImpl.java

```diff
@@ -0,0 +1,248 @@
+import static org.apache.cloudstack.resourcedetail.UserDetailVO.PasswordResetTokenExpiryDate;
+
+public class PasswordResetImpl extends ManagerBase implements PasswordReset, Configurable {
+
+    @Inject
+    private AccountManager accountManager;
+
+    @Inject
+    private UserDetailsDao userDetailsDao;
+
+    @Inject
+    private UserDao userDao;
+
+    private SMTPMailSender mailSender;
+
+    public static ConfigKey<String> PasswordResetMailTemplate =
+            new ConfigKey<>(ConfigKey.CATEGORY_ADVANCED, String.class,
+            "password.reset.mail.template", "Hello {{username}}!\n" +
+            "You have requested to reset your password. Please click the following link to reset your password:\n" +
+            "{{{reset_link}}}\n" +
+            "If you did not request a password reset, please ignore this email.\n" +
+            "\n" +
+            "Regards,\n" +
+            "The CloudStack Team",
+            "Password reset mail template. This uses mustache template engine. Available " +
+                    "variables are: username, firstName, lastName, resetLink, token",
+            true,
+            ConfigKey.Scope.Global);
+
+    @Override
+    public String getConfigComponentName() {
+        return PasswordResetImpl.class.getSimpleName();
+    }
+
+    @Override
+    public ConfigKey<?>[] getConfigKeys() {
+        return new ConfigKey<?>[]{PasswordResetTtl,
+                PasswordResetEmailSender,
+                PasswordResetSMTPHost,
+                PasswordResetSMTPPort,
+                PasswordResetSMTPUseAuth,
+                PasswordResetSMTPUsername,
+                PasswordResetSMTPPassword,
+                PasswordResetMailTemplate
+        };
+    }
+
+    @Override
+    public boolean configure(String name, Map<String, Object> params) throws ConfigurationException {
+        String smtpHost = PasswordResetSMTPHost.value();
+        Integer smtpPort = PasswordResetSMTPPort.value();
+        Boolean useAuth = PasswordResetSMTPUseAuth.value();
+        String username = PasswordResetSMTPUsername.value();
+        String password = PasswordResetSMTPPassword.value();
+
+        String namespace = "password.reset.smtp";
+
+        Map<String, String> configs = new HashMap<>();
+
+        configs.put(getKey(namespace, SMTPMailSender.CONFIG_HOST), smtpHost);
+        configs.put(getKey(namespace, SMTPMailSender.CONFIG_PORT), smtpPort.toString());
+        configs.put(getKey(namespace, SMTPMailSender.CONFIG_USE_AUTH), useAuth.toString());
+        configs.put(getKey(namespace, SMTPMailSender.CONFIG_USERNAME), username);
+        configs.put(getKey(namespace, SMTPMailSender.CONFIG_PASSWORD), password);
+
+        mailSender = new SMTPMailSender(configs, namespace);
+        return true;
+    }
+
+    private String getKey(String namespace, String config) {
+        return String.format("%s.%s", namespace, config);
+    }
+
+
+    public void setResetTokenAndSend(UserAccount userAccount) {
```

Comment:
Please add some logging in all the methods to log the flow in the management server logs.

Reply:
Added some logging.